src/sentry/static/sentry/app/views/eventsV2/savedQuery/index.tsx

```diff
@@ -286,9 +290,52 @@ class SavedQueryButtonGroup extends React.PureComponent<Props, State> {
     );
   }
 
+  renderButtonCreateAlert() {
+    const {eventView, organization, projects} = this.props;
+    // Must have exactly one project selected
+    const hasProjectError = eventView.project.length !== 1;
+    // Must have one or zero environments
+    const hasEnvironmentError = eventView.environment.length > 1;
+    // Must have event.type of error or transaction
+    const hasEventTypeError =
+      !eventView.query.includes('event.type:error') &&
+      !eventView.query.includes('event.type:transaction');
+    const project = projects.find(p => p.id === String(eventView.project[0]));
+    const isDisabled = hasProjectError || hasEnvironmentError || hasEventTypeError;
+
+    // TODO(scttcper): Implement real design for errors
+    let errorText = '';
+    if (hasProjectError) {
+      errorText = 'One project must be selected';
+    } else if (hasEnvironmentError) {
+      errorText = 'One or all environments is required';
+    } else if (hasEventTypeError) {
+      errorText = 'Either event.type:errors or event.type:transaction is required';
+    }
+
+    let newAlertUrl: object | undefined;
+    if (!isDisabled && project) {
+      const query = eventView.generateQueryStringObject();
+      query.createFromDiscover = 'true';
+      const pathname = `/settings/${organization.slug}/projects/${project.slug}/alerts/new/`;
+      newAlertUrl = {pathname, query};
+    }
+
+    return (
+      <Feature features={['create-from-discover']} organization={organization}>
+        <Tooltip title={errorText} disabled={!isDisabled}>
+          <Button disabled={isDisabled} to={newAlertUrl} icon={<IconSiren />}>
+            {t('Create alert')}
+          </Button>
+        </Tooltip>
+      </Feature>
+    );
+  }
```

Comment:
it'd be good to have tests for this

Reply:
added a light render test to this, its about to be changed to match the design and i'll cover with more tests